testData.js

```diff
@@ -5,11 +5,15 @@ var HTTP = require("./http");
 var Transcript = require("./transcript");
 
 var config = require("./config.json");
+const fs = require("fs");
+const path = require('path');
+
+const readFile = require('util').promisify(fs.readFile);
 
 class TestData {
 
     constructor(obj, query) {
-        this.name = (query && query.name) || (obj && obj.name) || (query && query.url.split('/').pop());
+        this.name = (query && query.name) || (obj && obj.name) || (query?.url?.split('/').pop()) || (query?.path?.split('/').pop());

```

Comment:
why not fix the whole line while you're at it?
`query?.name ?? obj?.name ?? (...)`

Reply:
The whole constructor should be refactored, but this is out of task's scope, and it's a matter of taste after all